In [130]:
import pandas as pd
import numpy as np
import re

只有13件有三位法官，其他都只有一位法官

In [1]:
with open(r'D:\Google雲端硬碟\NCKU\法經\法經報告\procedure\判決書路徑_終結案件_ver185.txt','r',encoding='utf-8-sig') as f:
    txt = f.read()

In [2]:
cases = txt.split('#\n')[1:]
len(cases) #共26928件

26928

In [6]:
case = cases[0]
print(case)

D:/Google雲端硬碟/NCKU/法經/法經報告/procedure/判決書\南投\南投_2019-01-02_投交簡_576.txt
南投地方法院!108!投交簡!576!公共危險!孫于淦!!!!!!!108!12!20!判決!1.00!!!!!0!0!0!0!!
1!108!12!20!!0!0!0!0!0!0!0!0!0!!0!0!0!0!!0!0!0!0!0!0!0!0!!!0!0!0!0!!0!!0!0!0!0!0!0!0!0!0!0!0!0!0!0!0!0!!!!!!!!!!!!!!0!0!!!!
1.1!刑法!185!3!!1!!!10806!有期徒刑!0002!!!1!0!0!0!0!0



In [7]:
case_features = case.split('!')
print(case_features)
len(case_features)

['D:/Google雲端硬碟/NCKU/法經/法經報告/procedure/判決書\\南投\\南投_2019-01-02_投交簡_576.txt\n南投地方法院', '108', '投交簡', '576', '公共危險', '孫于淦', '', '', '', '', '', '', '108', '12', '20', '判決', '1.00', '', '', '', '', '0', '0', '0', '0', '', '\n1', '108', '12', '20', '', '0', '0', '0', '0', '0', '0', '0', '0', '0', '', '0', '0', '0', '0', '', '0', '0', '0', '0', '0', '0', '0', '0', '', '', '0', '0', '0', '0', '', '0', '', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '', '', '', '', '', '', '', '', '', '', '', '', '', '0', '0', '', '', '', '\n1.1', '刑法', '185', '3', '', '1', '', '', '10806', '有期徒刑', '0002', '', '', '1', '0', '0', '0', '0', '0\n']


116

### data.csv

In [27]:
df = pd.read_csv(r'D:\Google雲端硬碟\NCKU\法經\法經報告\procedure\data.csv',encoding= 'utf-8-sig')

## 法官df合併進data.csv

In [28]:
#法官名,
judge=[]
for case in cases:
    case_features = case.split('!')
    judge.append( case_features[5] )
d = {'judge':judge}
df_j = pd.DataFrame(data=d)
#df_j = pd.Series(data = d)
type(df_j)

pandas.core.frame.DataFrame

In [35]:
df = pd.concat([df,df_j],axis = 'columns') #axis: {0/’index’, 1/’columns’}
df

file_name court        date case_kind  case_num judge
0      南投_2019-01-02_投交簡_576.txt    南投  2019-01-02       投交簡       576   孫于淦
1        南投_2019-01-07_埔交簡_2.txt    南投  2019-01-07       埔交簡         2   孫于淦
2      南投_2019-01-07_投交簡_521.txt    南投  2019-01-07       投交簡       521   楊國煜
3      南投_2019-01-07_投交簡_524.txt    南投  2019-01-07       投交簡       524   孫于淦
4      南投_2019-01-07_投交簡_531.txt    南投  2019-01-07       投交簡       531   黃俊明
...                          ...   ...         ...       ...       ...   ...
26923  高雄_2019-06-18_審交易_437.txt    高雄  2019-06-18       審交易       437   詹尚晃
26924  高雄_2019-07-16_審交易_528.txt    高雄  2019-07-16       審交易       528   詹尚晃
26925  高雄_2019-08-15_審交易_625.txt    高雄  2019-08-15       審交易       625   詹尚晃
26926  高雄_2019-08-20_審交易_652.txt    高雄  2019-08-20       審交易       652   詹尚晃
26927    高雄_2019-08-27_交易_49.txt    高雄  2019-08-27        交易        49   胡慧滿

[26928 rows x 6 columns]

In [36]:
df.to_csv('D:/Google雲端硬碟/NCKU/法經/法經報告/procedure/data.csv',encoding='utf-8-sig',index=False)
df.to_csv('D:/Google雲端硬碟/NCKU/法經/data.csv',encoding='utf-8-sig',index=False)

## 總刑期
先定位出 1!yyy!mm!dd! 該行 \
定位出 '應執行刑' \d{4} \
無的話往下找出 1.1!刑法!185! 該行 \
定位出 'yymm' \
將 'yymm' 轉成 int(mm) \
note: 1!的行數為被告數 只有33件有多個被告排除討論

In [114]:
case = cases[408]
print(case)

D:/Google雲端硬碟/NCKU/法經/法經報告/procedure/判決書\南投\南投_2019-06-18_審交易_84.txt
南投地方法院!108!審交易!84!公共危險等!葛耀陽!!!!!!!108!06!18!判決!1.00!!!!!0!0!0!0!!
1!108!06!18!!0!0!0!0!0!0!0!0!0!行簡式審!0!0!0!0!!0!0!0!0!0!0!0!0!!!0!0!0!0!!0!!0!0!0!0!0!0!0!0!0!0!0!0!0!0!0!0!!!!!!!0008!!!!!!!1!0!!!!
1.1!刑法!185!3!!1!!!10206!有期徒刑!0006!!!1!0!0!0!0!0!
1.1.1!累犯!
1.1!刑法!284!!!1!1!!!有期徒刑!0003!!!1!0!0!0!0!0!
1.1.2!自首



In [252]:
def jail_int(num):
    year = num[:2]
    month = num[2:]
    y=int(year) *12
    m = int(month)
    return y+m
#總樣本為26928-33-10 = 26885
match_1 = {}
match_2 = {}
file_names =[]
jails = []
for i,case in enumerate(cases): #共26928件
    lines = case.split('\n')
    #/ 排除多位被告案, 使每個case只會有一個 '1!10' 共33件
    count = 0
    for line in lines:
        if line[:2] == '1!': #因終結案跟判決書日期可能不同
            count = count + 1
            #print(i,case)
    if count > 1: 
        continue
    # /
    #先定位出 1!yyy!mm!dd! 該行, 確認過每個case裡lines[2]都有'1!10'
    
    input_text = lines[2]
    regex = '!\d{4}!'  
    pattern = re.compile(regex)
    num4 = re.findall(pattern, input_text) # 只有一組數字的有144件,沒數字的有26784件, 多組數字的不存在
    if len(num4) == 1: #得到最終判刑月數,共144件
        match_1[case] = num4[0]
        file_names.append( lines[0].split('\\')[-1] )
        jails.append( jail_int(num4[0][1:-1]) ) #拿掉驚嘆號
    else: #再往下探討, (1!下一行)最先出現1.1!的該行取出判刑月數
        for line in lines:
            if line[:4] == '1.1!': #得26751件
                input_text = line
                regex = '有期徒刑!\d{4}!'  
                pattern = re.compile(regex)
                num4_2 = re.findall(pattern, input_text) #無超過1的
                match_2[case] = num4_2
                if num4_2 ==[]: #無罪 不受理 免除其刑,共10件
                    break
                file_names.append( lines[0].split('\\')[-1] )
                jails.append( jail_int(num4_2[0][5:-1]) ) #拿掉有期徒刑!,!
                break

jails[10938] = 4
jails[18004] = 2
#兩個dist ,key為檔名,value為判刑月數, 各建dataframe 再垂直合併
#d = {'file_name':[r1,r2,...],'jail':[r1,r2,...]}


In [280]:
df = pd.read_csv(r'D:\Google雲端硬碟\NCKU\法經\法經報告\procedure\data.csv',encoding= 'utf-8-sig')
d = {'file_name':file_names,'jail':jails}
df_jail = pd.DataFrame(data=d)

In [ ]:
df.set_index(['file_name'],inplace = True)
df_jail.set_index(['file_name'],inplace = True)
#ex3.reset_index(inplace=True)

In [289]:
df_new

court        date case_kind  case_num judge  jail
南投_2019-01-02_投交簡_576.txt    南投  2019-01-02       投交簡       576   孫于淦   2.0
南投_2019-01-07_埔交簡_2.txt      南投  2019-01-07       埔交簡         2   孫于淦   3.0
南投_2019-01-07_投交簡_521.txt    南投  2019-01-07       投交簡       521   楊國煜   2.0
南投_2019-01-07_投交簡_524.txt    南投  2019-01-07       投交簡       524   孫于淦   5.0
南投_2019-01-07_投交簡_531.txt    南投  2019-01-07       投交簡       531   黃俊明   5.0
...                         ...         ...       ...       ...   ...   ...
高雄_2019-06-18_審交易_437.txt    高雄  2019-06-18       審交易       437   詹尚晃   8.0
高雄_2019-07-16_審交易_528.txt    高雄  2019-07-16       審交易       528   詹尚晃  16.0
高雄_2019-08-15_審交易_625.txt    高雄  2019-08-15       審交易       625   詹尚晃  10.0
高雄_2019-08-20_審交易_652.txt    高雄  2019-08-20       審交易       652   詹尚晃   5.0
高雄_2019-08-27_交易_49.txt      高雄  2019-08-27        交易        49   胡慧滿   7.0

[26928 rows x 6 columns]

In [291]:
df_new = pd.concat([df,df_jail],axis='columns')
df_new.reset_index(inplace=True)
df_new.to_csv('D:/Google雲端硬碟/NCKU/法經/法經報告/procedure/data.csv',encoding='utf-8-sig',index=False)
df_new.to_csv('D:/Google雲端硬碟/NCKU/法經/data.csv',encoding='utf-8-sig',index=False)

### -------------------------------------------------------------------------

In [254]:
len(match_1),len(match_2),len(file_names),len(jails)

(144, 26751, 26885, 26885)

In [251]:
print(file_names[18004],jails[10938])
for case in cases:
    lines = case.split('\n')
    if file_names[18004] in lines[0]:
        print(case)

臺中_2019-12-27_交易_887.txt 672
D:/Google雲端硬碟/NCKU/法經/法經報告/procedure/判決書\臺中\臺中_2019-12-27_交易_887.txt
臺中地方法院!108!交易!887!公共危險等!戰諭威!!!!!!!108!12!27!判決!1.00!!!!!0!0!0!0!!
1!108!12!27!!0!0!0!0!0!0!0!0!0!認罪協商!1!0!0!0!8000!0!0!0!0!0!0!0!0!!!0!0!0!0!!0!!0!0!0!0!0!0!0!0!0!0!0!0!0!0!0!0!!!!!!!!!!!!!!0!0!!!!
1.1!刑法!185!3!!1!!!10806!有期徒刑!0002!!10,000!1!0!0!0!0!0!
1.1!刑法!284!!!1!1!!!不受理!!!!0!0!0!0!0!0



In [253]:
#確認file_name,jail有效性
for i,t in enumerate(jails):
    #if i == 20000:
     #   break
    if t>12:
        print(i,t)

320 14
505 13
546 13
3913 16
4810 14
5179 13
5213 20
5698 16
5850 16
7330 13
10063 15
10375 20
11443 22
11669 14
11699 14
13290 15
13463 14
14164 24
14165 13
14367 13
15076 14
15252 14
15492 26
15787 17
17060 13
17488 26
19091 18
19164 18
19583 14
20565 14
20848 14
21437 14
21496 14
21604 20
22592 13
22683 16
22825 16
24326 14
24524 14
24623 15
25044 19
25297 30
25847 14
26184 13
26652 15
26881 16


In [161]:
len(match_1),len(non_match),len(match_muti)

(144, 26751, 0)

In [162]:
for i,case in enumerate(match_1):
    print(case,match_1[case])
    if i == 50:
        break

312 D:/Google雲端硬碟/NCKU/法經/法經報告/procedure/判決書\南投\南投_2019-05-13_審交易_42.txt
南投地方法院!108!審交易!42!公共危險等!張家豪!!!!!!!108!05!13!判決!1.00!!!!!0!0!0!0!!
1!108!05!13!!0!0!0!0!0!0!0!0!0!行簡式審!0!0!0!0!!0!0!0!0!0!0!0!0!!!0!0!0!0!!0!!0!0!0!0!0!0!0!0!0!0!0!0!0!0!0!0!!!!!!!0007!!!!!!!0!0!!!!
1.1!刑法!185!3!!1!!!10206!有期徒刑!0005!!!1!0!0!0!0!0!
1.1!刑法!284!!!1!1!!!有期徒刑!0003!!!1!0!0!0!0!0

408 D:/Google雲端硬碟/NCKU/法經/法經報告/procedure/判決書\南投\南投_2019-06-18_審交易_84.txt
南投地方法院!108!審交易!84!公共危險等!葛耀陽!!!!!!!108!06!18!判決!1.00!!!!!0!0!0!0!!
1!108!06!18!!0!0!0!0!0!0!0!0!0!行簡式審!0!0!0!0!!0!0!0!0!0!0!0!0!!!0!0!0!0!!0!!0!0!0!0!0!0!0!0!0!0!0!0!0!0!0!0!!!!!!!0008!!!!!!!1!0!!!!
1.1!刑法!185!3!!1!!!10206!有期徒刑!0006!!!1!0!0!0!0!0!
1.1.1!累犯!
1.1!刑法!284!!!1!1!!!有期徒刑!0003!!!1!0!0!0!0!0!
1.1.2!自首

495 D:/Google雲端硬碟/NCKU/法經/法經報告/procedure/判決書\南投\南投_2019-07-18_審交易_94.txt
南投地方法院!108!審交易!94!公共危險!張家豪!!!!!!!108!07!18!判決!1.00!!!!!0!0!0!0!!
1!108!07!18!!0!0!0!0!0!0!0!0!0!行簡式審!0!0!0!0!!0!0!0!0!0!0!0!0!!!0!0!0!0!!0!!0!0!0!0!0!0!0!0!0!0!0!0!0!0!0!0!!!!!!!0

In [149]:
#### 判斷是否存在
case = cases[408]
targ = []
for i,case in enumerate(cases):
    lines = case.split('\n')
    #先定位出 1!yyy!mm!dd! 該行
    if '1.1!刑法!' not in lines[3]: #因終結案跟判決書日期可能不同
        #print(i,case)
        targ.append(case)
len(targ)


2

# 排除多個被告

In [167]:
# 看多個1.1!
#string.startswith("目標字符")

targ = []
for i,case in enumerate(cases):
    lines = case.split('\n')
    #/ 排除多位被告案, 使每個case只會有一個 '1!10'
    count = 0
    for line in lines:
        if line[:4] == '1.1!': #因終結案跟判決書日期可能不同
            count = count + 1
            #print(i,case)
    if count > 1: 
        continue
    # /
    #先定位出 1!yyy!mm!dd! 該行
 

116

In [168]:
for i in targ:
    print(i)

D:/Google雲端硬碟/NCKU/法經/法經報告/procedure/判決書\南投\南投_2019-05-20_埔原交簡_31.txt
南投地方法院!108!埔原交簡!31!公共危險等!孫于淦!!!!!!!108!05!20!判決!1.00!!!!!0!0!0!0!!
1!108!05!20!!0!0!0!0!0!0!0!0!0!!0!0!0!0!!0!0!0!0!0!0!0!0!!!0!0!0!0!!0!!0!0!0!0!0!0!0!0!0!0!0!0!0!0!0!0!!!!!!!!!!!!!!0!0!!!!
1.1!刑法!185!3!!1!!!10206!有期徒刑!0002!!!1!0!0!0!0!0!
1.1!刑法!284!!!1!1!!!拘役!!30!!1!0!0!0!0!0!
1.1.2!自首

D:/Google雲端硬碟/NCKU/法經/法經報告/procedure/判決書\南投\南投_2019-07-18_交易_10.txt
南投地方法院!108!交易!10!公共危險等!楊國煜!!!!!!!108!07!18!判決!1.00!!!!!0!0!0!0!!
1!108!07!18!!0!0!0!0!0!0!0!0!0!!0!0!0!0!!0!0!0!0!0!0!0!0!!!0!0!0!0!!0!!0!0!0!0!0!0!0!0!0!0!0!0!0!0!0!0!!!!!!!!!!!!!!0!0!!!!
1.1!刑法!185!3!!1!!!10806!有期徒刑!0005!!!1!0!0!0!0!0!
1.1!刑法!284!!!1!1!!!不受理!!!!0!0!0!0!0!0

D:/Google雲端硬碟/NCKU/法經/法經報告/procedure/判決書\南投\南投_2019-09-27_埔交簡_127.txt
南投地方法院!108!埔交簡!127!公共危險等!葉峻石!!!!!!!108!09!27!判決!1.00!!!!!0!0!0!0!!
1!108!09!27!!0!0!0!0!0!0!0!0!0!!0!0!0!0!!0!0!0!0!0!0!0!0!!!0!0!0!0!!0!!0!0!0!0!0!0!0!0!0!0!0!0!0!0!0!0!!!!!!!!!!!!!!0!0!!!!
1.1!刑法!185!3!!1!!!10806!有期徒刑!0005!

In [246]:
file ='D:/Google雲端硬碟/NCKU/法經/法經報告/procedure/判決書\桃園\桃園_2019-02-23_壢交簡_107.txt'
with open(file,'r',encoding='utf-8-sig') as f:
    ex_txt = f.read()
print(ex_txt)

臺灣桃園地方法院刑事簡易判決　　　108年度壢交簡字第107號

聲　請　人　臺灣桃園地方檢察署檢察官

被　　　告　徐建復





上列被告因公共危險案件，經檢察官聲請以簡易判決處刑（107

年度速偵字第6160號），本院判決如下：

    主  文

徐建復駕駛動力交通工具而有吐氣所含酒精濃度達每公升零點二

五毫克以上情形，處有期徒刑肆月，如易科罰金，以新臺幣壹仟

元折算壹日。

    事實及理由

一、本案犯罪事實及證據均引用檢察官聲請簡易判決處刑書之記

    載（如附件）。

二、核被告徐建復所為，係犯刑法第185 條之3 第1 項第1 款之

    駕駛動力交通工具而有吐氣所含酒精濃度達每公升0.25毫克

    以上情形之罪。爰審酌被告於服用酒類後，吐氣所含酒精濃

    度高達每公升0.41毫克之狀態下，仍執意駕駛自用小客車上

    路，除不顧己身安全外，更漠視往來公眾之人身、財產安全

    ，殊值非難；惟念及被告於犯後坦承犯行，態度尚佳，又本

    次違法行為並未肇生交通事故及其犯罪動機、目的等一切情

    狀，量處如主文所示之刑，並諭知易科罰金之折算標準，以

    示懲儆。

三、據上論斷，應依刑事訴訟法第449 條第1 項前段、第3 項、

    第454 條第2 項，刑法第185 條之3 第1 項第1 款、第41條

    第1 項前段，刑法施行法第1 條之1 第1 項，逕以簡易判決

    處刑如主文。

四、如不服本判決，應於判決送達後10日內，向本院提出上訴狀

    ，上訴於本院第二審合議庭（應附繕本）。

本案業經檢察官白覲毓聲請簡易判決處刑

中    華    民    國    108    年   2    月    23     日

                  刑事第八庭  法  官  呂宜臻

以上正本證明與原本無異。

如不服本判決，應於判決送達後10日內向本院提出上訴狀（應附

繕本）。

                              書記官  王小萍

中    華    民    國   108    年    3     月    1     日

附錄本案論罪科刑依據之法條：

中華民國刑法第185條之3

駕駛動力交通工具而有下列情形之

In [111]:
#短的
print(cases[0] )

D:/Google雲端硬碟/NCKU/法經/法經報告/procedure/判決書\南投\南投_2019-01-02_投交簡_576.txt
南投地方法院!108!投交簡!576!公共危險!孫于淦!!!!!!!108!12!20!判決!1.00!!!!!0!0!0!0!!
1!108!12!20!!0!0!0!0!0!0!0!0!0!!0!0!0!0!!0!0!0!0!0!0!0!0!!!0!0!0!0!!0!!0!0!0!0!0!0!0!0!0!0!0!0!0!0!0!0!!!!!!!!!!!!!!0!0!!!!
1.1!刑法!185!3!!1!!!10806!有期徒刑!0002!!!1!0!0!0!0!0



### 找出有8行以上的case

In [105]:
#找出有5行以上的case，看總刑期會寫在哪
dic = {}
long= []
for i,case in enumerate( cases ):
    if len(case.split('\n')) >7:
        long.append(case)
        dic[i] =case
len(long)

123

In [108]:
for i,case in enumerate(dic):
    print(case,dic[case])
    if i == 5:
        break

408 D:/Google雲端硬碟/NCKU/法經/法經報告/procedure/判決書\南投\南投_2019-06-18_審交易_84.txt
南投地方法院!108!審交易!84!公共危險等!葛耀陽!!!!!!!108!06!18!判決!1.00!!!!!0!0!0!0!!
1!108!06!18!!0!0!0!0!0!0!0!0!0!行簡式審!0!0!0!0!!0!0!0!0!0!0!0!0!!!0!0!0!0!!0!!0!0!0!0!0!0!0!0!0!0!0!0!0!0!0!0!!!!!!!0008!!!!!!!1!0!!!!
1.1!刑法!185!3!!1!!!10206!有期徒刑!0006!!!1!0!0!0!0!0!
1.1.1!累犯!
1.1!刑法!284!!!1!1!!!有期徒刑!0003!!!1!0!0!0!0!0!
1.1.2!自首

528 D:/Google雲端硬碟/NCKU/法經/法經報告/procedure/判決書\南投\南投_2019-08-05_審交易_107.txt
南投地方法院!108!審交易!107!公共危險!張家豪!!!!!!!108!08!05!判決!1.00!!!!!0!0!0!0!!
1!108!08!05!!0!0!0!0!0!0!0!0!0!行簡式審!0!0!0!0!!0!0!0!0!0!0!0!0!!!0!0!0!0!!0!!0!0!0!0!0!0!0!0!0!0!0!0!0!0!0!0!!!!!!!0006!!!!!!!1!0!!!!
1.1!刑法!185!3!!1!!!10806!有期徒刑!0004!!!1!0!0!0!0!0!
1.1.1!累犯!
1.1!刑法!185!3!!1!!!10806!有期徒刑!0003!!!1!0!0!0!0!0!
1.1.1!累犯

739 D:/Google雲端硬碟/NCKU/法經/法經報告/procedure/判決書\南投\南投_2019-09-27_埔交簡_83.txt
南投地方法院!108!埔交簡!83!公共危險等!蔡志明!!!!!!!108!09!27!判決!1.00!!!!!0!0!0!0!!
1!108!09!27!!0!0!0!0!0!0!0!0!0!!0!0!0!0!!0!0!0!0!0!0!0!0!!!0!0!0!0!!0!!0!0!0!0!0!0!0!0!

In [100]:
ex=  long[29]
ex.split('\n')
print(ex)

D:/Google雲端硬碟/NCKU/法經/法經報告/procedure/判決書\彰化\彰化_2019-11-27_交簡_1572.txt
彰化地方法院!108!交簡!1572!公共危險!黃玉齡!!!!!!!108!11!27!判決!1.00!!!!!0!0!0!0!!
1!108!11!27!!0!0!0!0!0!0!0!0!0!!0!0!0!0!!0!0!0!0!0!0!0!0!!!0!0!0!0!!0!!0!0!0!0!0!0!0!0!0!0!0!0!0!0!0!0!!!!!!!0008!!!!!!!1!0!!!!
1.1!刑法!185!3!!1!!!10806!有期徒刑!0005!!!1!0!0!0!0!0!
1.1.1!累犯!
1.1!刑法!185!3!!1!!!10806!有期徒刑!0004!!!1!0!0!0!0!0!
1.1.1!累犯



In [101]:
with open(ex.split('\n')[0],'r',encoding='utf-8-sig') as f:
    ex_txt = f.read()
print(ex_txt)

臺灣彰化地方法院刑事簡易判決　　　 108年度交簡字第1572號

　　　　　　　　　　　　　　　　　 108年度交簡字第1663號

公　訴　人　臺灣彰化地方檢察署檢察官

被　　　告　江延祚







上列被告因公共危險案件，經檢察官提起公訴（108 年度偵字第

3390號、108 年度偵字第7124號），被告自白犯罪，本院認宜以

簡易判決處刑，爰逕以簡易判決處刑如下：

     主  文

江延祚駕駛動力交通工具而吐氣所含酒精濃度達每公升零點二五

毫克以上（兩罪），均累犯，各處有期徒刑肆月、伍月，如易科

罰金，均以新臺幣壹仟元折算壹日。應執行有期徒刑捌月，如易

科罰金，以新臺幣壹仟元折算壹日。

    事實及理由

一、犯罪事實：

（一）江延祚於民國108 年3 月26日下午8 時許，在彰化縣永靖

      鄉某處飲用高粱酒後，已處於不能安全駕駛狀態，仍駕駛

      車牌號碼0000-00 號自用小客車上路。嗣於108 年3 月27

      日0 時35分許，行經彰化縣員林市員林大道2 段與明倫路

      口，因行車不穩為警攔查，並於同日凌晨1 時7 分許，測

      得其呼氣酒精濃度值達每公升0.55毫克。

（二）又於108 年7 月4 日22時許，在其友人位於彰化縣員林市

      舊愛買購物中心附近之友人住處，飲用補藥酒後，已處於

      不能安全駕駛狀態，仍駕駛車牌號碼0000-00 號自用小客

      車上路。嗣於翌（5 ）日0 時47分許，行經彰化縣○○市

      ○○路0 段000 巷0 號前，因逆向行駛為警攔查，並測得

      其呼氣酒精濃度值達每公升0.69毫克。

二、前開犯罪事實，業據被告江延祚於警詢、偵查中及本院審理

    時坦承不諱，復有彰化縣警察局道路交通事故當事人酒精測

    定紀錄表二份、彰化縣○○○○○○○道路○○○○○○○

    ○○○○○○○○號查詢汽車駕駛人及車號查詢汽車車籍資

    料等件在卷可參，本案事證明確，被告兩次酒後駕車之犯行

    堪以認定。

三、論罪科刑：

（一）核被告兩次酒後駕車之行為，均是犯刑法第185 條之3 第

      1 項第1 款之吐氣所含酒精濃度達每公升0.25毫克以上而

      

##

In [ ]:
#累犯寫的位置跟意義

### ------------------------------------------------------------------------------------------------------------

In [9]:
d = {'col1': [1, 2], 'col2': [3, 4]}
ex1 = pd.DataFrame(data=d)
ex1

col1  col2
0     1     3
1     2     4

In [10]:
ex2 = pd.DataFrame(np.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]]),
                   columns=['a', 'b', 'c'])
#ex2.to_csv('test.csv',index=False)
ex2

a  b  c
0  1  2  3
1  4  5  6
2  7  8  9

In [3]:
data = []
#法官名,
for case in cases:
    row_data=[ ]
    case_features = case.split('!')
    judge = case_features[5]
    row_data.append(judge)
    data.append(row_data)

In [14]:
#確認有無第二位法官
second_man = {}
for i,case in enumerate(cases):
    if case == '':
        continue
    case_features = case.split('!')
    if case_features[6] == '':
        continue
    second_man[i]= case_features[6]
print(len(second_man) )
print(second_man )

13
{958: '黃益茂', 4026: '黃永勝', 9350: '劉安榕', 10187: '楊數盈', 10536: '賴淑敏', 12850: '張宏任', 14383: '林揚奇', 17241: '胡宜如', 17973: '唐中興', 24163: '黃英豪', 25319: '魏宏安', 25508: '魏宏安', 26757: '蕭于哲'}


In [15]:
#確認有無第三位法官
third_man = {}
for i,case in enumerate(cases):
    if case == '':
        continue
    case_features = case.split('!')
    if case_features[7] == '':
        continue
    third_man[i]= case_features[7]
print(len(third_man) )
print(third_man )

13
{958: '葛耀陽', 4026: '張淑華', 9350: '林正忠', 10187: '江宜穎', 10536: '王靜慧', 12850: '林姿秀', 14383: '郭育秀', 17241: '廖慧娟', 17973: '黃佳琪', 24163: '許芳瑜', 25319: '王瀅婷', 25508: '王瀅婷', 26757: '陳育良'}


In [21]:
for i in third_man:
    print(cases[i])
    with open(cases[i].split('\n')[0],'r',encoding='utf-8-sig') as f:
        print(f.read() )

D:/Google雲端硬碟/NCKU/法經/法經報告/procedure/判決書\南投\南投_2019-12-04_審交易_214.txt
南投地方法院!108!審交易!214!公共危險等!葉峻石!黃益茂!葛耀陽!!!!!108!12!04!判決!1.00!!!!!0!0!0!0!!
1!108!12!04!!0!0!0!0!0!0!0!0!0!行簡式審!0!0!0!0!!0!0!0!0!0!0!0!0!!!0!0!0!0!!0!!0!0!0!0!0!0!0!0!0!0!0!0!0!0!0!0!!!!!!!!!!!!!!0!0!!!!
1.1!刑法!185!3!!1!!!10806!有期徒刑!0004!!!1!0!0!0!0!0!
1.1!刑法!284!!!1!1!!!不受理!!!!0!0!0!0!0!0

臺灣南投地方法院刑事判決　　　　　108年度審交易字第214號

公　訴　人　臺灣南投地方檢察署檢察官

被　　　告　江山城





上列被告因公共危險案件，經檢察官提起公訴（108 年度偵字第

1888號），因被告於準備程序中就被訴事實為有罪之陳述，經告

以簡式審判程序之旨，並聽取當事人意見後，本院合議庭裁定由

受命法官獨任進行簡式審判程序，判決如下：

    主  文

江山城駕駛動力交通工具而吐氣所含酒精濃度達每公升零點二五

毫克以上，處有期徒刑肆月，如易科罰金，以新臺幣壹仟元折算

壹日。

    事實及理由

一、犯罪事實：

  江山城於民國108 年4 月21日14時許，在南投縣南投市奉天

    宮附近某友人住處飲用啤酒後，竟仍基於酒後駕駛動力交通

    工具之犯意，於同日14時51分許，駕駛車牌號碼0000-00 號

    自用小客車上路。嗣因其於同日14時56分許，行經南投縣○

    ○市○○路0 號前時，撞及行人蔡秀美致其受骨折等傷害（

    過失傷害罪嫌部分，業經撤回告訴，由本院另行審結），經

    警據報前往現場處理，並於同日15時18分許對江山城施以呼

    氣酒精濃度測試，測得其吐氣酒精濃度為每公升0.93毫克，

    而查悉上情。

  案經南投縣政府警察局南投分局報告臺灣南投地方檢察署檢

    察官偵查起訴。

二、證據名稱：

  被告江山城於警詢、偵訊、本院準備程序及審理

In [17]:
cases[4026] 

'D:/Google雲端硬碟/NCKU/法經/法經報告/procedure/判決書\\宜蘭\\宜蘭_2019-07-16_交易_95.txt\n宜蘭地方法院!108!交易!95!公共危險!陳盈孜!黃永勝!張淑華!!!!!108!07!16!判決!1.00!!!!!0!0!0!0!!\n1!108!07!16!!0!0!0!0!0!0!0!0!0!!0!0!0!0!!0!0!0!0!0!0!0!0!!!0!0!0!0!!0!!0!0!0!0!0!0!0!0!0!0!0!0!0!0!0!0!!!!!!!!!!!!!!1!0!!!!\n1.1!刑法!185!3!!1!!!10806!有期徒刑!0008!!!0!0!0!0!0!0!\n1.1.1!累犯\n'

In [ ]:
case